drivers/adc/ad7779/ad7779.c

```diff
@@ -598,10 +633,27 @@ int32_t ad7779_get_dec_rate(ad7779_dev *dev,
 			    uint16_t *int_val,
 			    uint16_t *dec_val)
 {
-	*int_val = dev->dec_rate_int;
-	*dec_val = dev->dec_rate_int;
+	int ret = 0;
+	uint8_t reg_data;
+
+	if (!dev->read_from_cache) {
 
-	return 0;
+		ret |= ad7779_spi_int_reg_read(dev, AD7779_REG_SRC_IF_LSB, &reg_data);
```

Comment:
general aspect. even though probably the driver is old and this approach is used throughout the driver, let's perform `ret` checks after each function call, instead of propagating the return codes throughout the entire function.

Reply:
Noted. ret check has been added wherever applicable